lib/src/arithmetic/values/fixed_point_value.dart

```diff
@@ -143,18 +143,58 @@ class FixedPointValue implements Comparable<FixedPointValue> {
     return compareTo(other) == 0;
   }
 
-  /// Constructs [FixedPointValue] of a Dart [double] rounding away from zero.
+  /// Return a string representation of FloatingPointValue.
+  ///  return sign, exponent, mantissa as binary strings.
+  @override
+  String toString() =>
+      "(${signed ? '${value[-1].toString(includeWidth: false)} ' : ''}"
+      "${(m > 0) ? '${value.slice(m + n - 1, n).bitString} ' : ''}"
+      '${value.slice(n - 1, 0).toString(includeWidth: false)})';
+
+  /// Return true if double [val] be stored in FixedPointValue with [m] and [n]
+  /// lengths.
+  static bool canStore(double val,
+      {required bool signed, required int m, required int n}) {
+    final w = signed ? 1 + m + n : m + n;
+    if (val.isFinite) {
+      final bigIntegerValue = BigInt.from(val * pow(2, n));
+      final negBigIntegerValue = BigInt.from(-val * pow(2, n));
+      final l = (val < 0.0)
+          ? max(bigIntegerValue.bitLength, negBigIntegerValue.bitLength)
+          : bigIntegerValue.bitLength;
+      return l <= w;
+    }
+    return false;
+  }
+
+  /// Constructs [FixedPointValue] from a Dart [double] rounding away from zero.
   factory FixedPointValue.ofDouble(double val,
```

Comment:
should this have a "rounding mode" option instead of having two separate functions?

Reply:
That's a good idea.  I will need to refactor ofDouble, ofDoubleUnrounded in both FixedPointValue and FloatingPointValue.

I'd like to do that after figuring out removing factory methods and getting overrides working in FPV and FXPV to kill all subtype references in the base classes and as much boilerplate as possible in the subclasses.

Reply:
Can I file this as an issue?  Could be an 'easy' one.

Reply:
Filed as Issue #173